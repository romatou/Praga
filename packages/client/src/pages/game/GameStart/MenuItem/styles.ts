import { CSSProperties } from 'react'

export const itemStyles = {
  width: '216px',
  borderRadius: '0px 6px 6px 0px',
  boxShadow: '4px 4px 4px 4px rgba(0, 0, 0, 0.25)',
  transition: 'all .2s ease',
  '&:hover': {
    width: '248px',
  },
}

export const linkStyles: CSSProperties = {
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  flexDirection: 'column',
  gap: '16px',
  padding: '24px 32px',
  textDecoration: 'none',
}
