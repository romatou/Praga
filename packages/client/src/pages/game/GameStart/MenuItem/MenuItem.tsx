import { Link } from 'react-router-dom'
import { Box, Typography } from '@mui/material'

import { itemStyles, linkStyles } from './styles'

interface MenuItemProps {
  title: string
  src: string
  alt: string
  to: string
}

export default function MenuItem({ title, src, alt, to }: MenuItemProps) {
  if (!to) return null

  return (
    <Box component="li" sx={itemStyles}>
      <Link to={to} style={linkStyles}>
        <Box component="img" {...{ src, alt }} />
        <Typography component="span">{title}</Typography>
      </Link>
    </Box>
  )
}
