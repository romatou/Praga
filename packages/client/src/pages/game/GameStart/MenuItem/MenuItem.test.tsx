import PersonIcon from '@mui/icons-material/Person'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'

import MenuItem from './MenuItem'

const title = 'Title'

describe('MenuItem component', () => {
  it('MenuItem renders', () => {
    render(
      <BrowserRouter>
        <MenuItem
          Icon={<PersonIcon sx={{ fontSize: 40 }} color="secondary" />}
          title={title}
          to="/"
        />
      </BrowserRouter>
    )

    expect(screen.getByRole('listitem')).toBeInTheDocument()
    expect(screen.getByText(title)).toBeInTheDocument()
  })

  it('MenuItem not renders without param - "to"', () => {
    render(
      <BrowserRouter>
        <MenuItem
          Icon={<PersonIcon sx={{ fontSize: 40 }} color="secondary" />}
          title={title}
          to=""
        />
      </BrowserRouter>
    )

    expect(screen.queryByRole('listitem')).toBeNull()
  })
})
