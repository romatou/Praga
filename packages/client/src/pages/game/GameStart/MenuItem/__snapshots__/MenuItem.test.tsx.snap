// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component MenuItem empty snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MenuItem component MenuItem snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="MuiBox-root css-4wnea4"
      >
        <a
          href="/"
          style="display: flex; justify-content: center; align-items: center; flex-direction: column; gap: 16px; padding: 24px 32px; text-decoration: none;"
        >
          <img
            alt=""
            class="MuiBox-root css-0"
            src=""
          />
          <span
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Title
          </span>
        </a>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="MuiBox-root css-4wnea4"
    >
      <a
        href="/"
        style="display: flex; justify-content: center; align-items: center; flex-direction: column; gap: 16px; padding: 24px 32px; text-decoration: none;"
      >
        <img
          alt=""
          class="MuiBox-root css-0"
          src=""
        />
        <span
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Title
        </span>
      </a>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
